docs/source/cpp/tables.rst

```diff
@@ -77,6 +77,18 @@ has a schema which must match its arrays' datatypes.
 Record batches are a convenient unit of work for various serialization
 and computation functions, possibly incremental.
 
+.. image:: tables-versus-record-batches.svg
+   :alt: A graphical representation of an Arrow Table and a Record Batch, with
+         structure as described in text above.
+
+Record batches can be sent between implementations, such as via 
+:ref:`IPC <format-ipc>` or
+via the :doc:`C Data Interface <../format/CDataInterface>`. Tables and 
+chunked arrays, on the other hand, are concepts in the C++ implementation (and
+its bindings), not in the Arrow format itself, so they aren't directly portable.
```

Comment:
Can we link up to the list of bindings somewhere, or should we expect readers to know if they're using a binded or fully-implemented implementation?

Reply:
This is within the C++ user guide, so I'm not too worried about Python or R users reading this and not being sure if this applies to them. Presumably they would only be on this page if they already knew!

But we should create a diagram on the Docs homepage that shows the relationship between implementations. I'll leave that as a separate ticket though.

Reply:
Let's perhaps remove the mention of bindings then?